search.go

```diff
@@ -0,0 +1,43 @@
+package filter
+
+import (
+	"gorm.io/gorm"
+)
+
+type Search struct {
+	Fields []string
+	Query  string
+}
+
+func (s *Search) Scopes(settings *Settings, modelIdentity *modelIdentity) func(*gorm.DB) *gorm.DB {
```

Comment:
```suggestion
func (s *Search) Scope(settings *Settings, modelIdentity *modelIdentity) func(*gorm.DB) *gorm.DB {
```
This function only returns one scope so it can be called `Scope` instead of `Scopes`.